feature-libs/cart/added-to-cart-toast/core/services/added-to-cart-toast.service.ts

```diff
@@ -0,0 +1,50 @@
+import { Injectable } from '@angular/core';
+import { Product, ProductService } from '@spartacus/core';
+import { Observable, of } from 'rxjs';
+import { CartToastItem } from '../models/added-to-cart-toast.model';
+
+@Injectable({ providedIn: 'root' })
+export class AddedToCartToastService {
+  cartToastItems: CartToastItem[] = [];
+
+  constructor(protected productService: ProductService) {}
+
+  getToasts(): Observable<CartToastItem[]> {
+    return of(this.cartToastItems);
+  }
+
+  addToast(quantityAdded: number, product: Product, containerClass: string) {
+    const newCartToastItem: CartToastItem = {
+      productName: product.name,
+      quantity: quantityAdded,
+      baseClass: containerClass,
+      unitPrice: product.price?.formattedValue,
+      images: product.images,
+    };
+    const index = this.cartToastItems.push(newCartToastItem) - 1;
+    setTimeout(() => {
```

Comment:
Why you need to use setTimeout here?

Reply:
I'm using this to account for the slide in animation. I could perhaps move this logic to the component if it's possible.